core/src/dht/dht_inner_reducers.rs

```diff
@@ -0,0 +1,135 @@
+///
+/// Inner DHT reducers are not pure functions but rather functions designed to make the required
+/// mutations to a newly cloned DhtState object. Unlike the reducers they do not need a specific signature.
+/// The should have a signature similar to
+///
+/// `reduce_some_thing_inner(store: &mut DhtStore, <other required data>) -> HcResult<someReturnType>`
+///
+/// It is up to the calling reducer function whether the new state object should be kept and what to do with the return value
+///
+use crate::dht::dht_store::DhtStore;
+use holochain_core_types::{
+    cas::content::{Address, AddressableContent},
+    crud_status::{create_crud_link_eav, create_crud_status_eav, CrudStatus},
+    eav::{Attribute, EaviQuery, EntityAttributeValueIndex, IndexFilter},
+    entry::Entry,
+    error::{HcResult, HolochainError},
+    link::Link,
+};
+use std::{collections::BTreeSet, convert::TryFrom, str::FromStr};
+
+pub(crate) enum LinkModification {
+    Add,
+    Remove,
+}
+
+/// Used as the inner function for both commit and hold reducers
+pub(crate) fn reduce_store_entry_inner(store: &mut DhtStore, entry: &Entry) -> HcResult<()> {
+    match (*store.content_storage().write()?).add(entry) {
+        Ok(()) => create_crud_status_eav(&entry.address(), CrudStatus::Live).map(|status_eav| {
+            (*store.meta_storage().write()?)
+                .add_eavi(&status_eav)
+                .map(|_| ())
+                .map_err(|e| format!("err/dht: dht::reduce_hold_entry() FAILED {:?}", e).into())
+        })?,
+        Err(e) => Err(format!("err/dht: dht::reduce_hold_entry() FAILED {:?}", e).into()),
+    }
+}
+
+pub(crate) fn reduce_add_remove_link_inner(
+    store: &mut DhtStore,
+    link: &Link,
+    link_modification: LinkModification,
+) -> HcResult<Address> {
+    match (*store.content_storage().read()?).contains(link.base())? {
+        true => {
```

Comment:
I think it might be clearer to use an if / else statement here? What do you think about that suggestion?

Reply:
Great suggestion. I made the change and it is more readable now